cocotb/handle.py

```diff
@@ -656,13 +656,16 @@ def _set_value(self, value, call_sim):
         """
         value, set_action = self._check_for_set_action(value)
 
-        if isinstance(value, int) and value < 0x7fffffff and len(self) <= 32:
-            call_sim(self._handle.set_signal_val_long, set_action, value)
+        if isinstance(value, int):
+            if value >= -0x80000000 and value <= 0x7fffffff and len(self) <= 32:
+                call_sim(self._handle.set_signal_val_long, set_action, value)
+            else:
+                value = value.to_bytes((len(self)+7)//8, byteorder='big', signed=value<0)
+                call_sim(self._handle.set_signal_val_bytes, set_action, value)
             return
+
         if isinstance(value, ctypes.Structure):
             value = BinaryValue(value=cocotb.utils.pack(value), n_bits=len(self))
-        elif isinstance(value, int):
-            value = BinaryValue(value=value, n_bits=len(self), bigEndian=False)
```

Comment:
Should we get rid of this, or change the `binaryRepresentation` to two's complement? It works the same for unsigned numbers, but supports negative numbers in the representation used by all (fight me C standards committee) modern processor archs.

Reply:
Nvm, we probably should get rid of it. But I don't think Eric will agree.

Reply:
This case should be covered by the else clause on line 662.
Essentially we could achieve the same by using the ```binaryRepresentation``` for negative values and default for positive values

Reply:
I think there is some confusion here with big/little endianness vs ascending/descending vector range. 
```set_signal_value_long()``` is used for setting vectors < 32-bit and assumes that the vector we're setting has a "descending" range. I wanted to ensure that the behaviour of setting vectors > 32-bit using ```set_signal_value_bytes()``` is consistent with how it is for setting vectors < 32-bit.

Here's a code snippet that illustrates that we want to use big (network) endian to ensure the bit pattern remains as intended left to right and avoid any extra work with reordering of the bytes in the C++ code. 

```python
In [1]: N=0x0123456789abcdef

In [2]: N.to_bytes(10, byteorder="big", signed=False).hex()
Out[2]: '00000123456789abcdef'

In [3]: N.to_bytes(10, byteorder="little", signed=False).hex()
Out[3]: 'efcdab89674523010000'

In [4]: N.to_bytes(10, byteorder="big", signed=False).hex()[::-1]
Out[4]: 'fedcba98765432100000'

In [5]: N.to_bytes(10, byteorder="little", signed=False).hex()[::-1]
Out[5]: '00001032547698badcfe'
```